core/ir/riscv64/opcode_api.h

```diff
@@ -0,0 +1,89 @@
+/* **********************************************************
+ * Copyright (c) 2022 Rivos, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Rivos, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL RIVOS, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#ifndef _DR_IR_OPCODES_RISCV64_H_
+#define _DR_IR_OPCODES_RISCV64_H_ 1
+
+/****************************************************************************
+ * OPCODES
+ */
+/**
+ * @file dr_ir_opcodes_arm.h
+ * @brief Instruction opcode constants for RISC-V.
+ */
+/** Opcode constants for use in the instr_t data structure. */
+enum {
+    /*   0 */ OP_INVALID,
+    /* NULL, */ /**< INVALID opcode */
+    /*   1 */ OP_UNDECODED,
+    /* NULL, */ /**< UNDECODED opcode */
+    /*   2 */ OP_CONTD,
+    /* NULL, */ /**< CONTD opcode */
+    /*   3 */ OP_LABEL,
+    /* NULL, */ /**< LABEL opcode */
+
+    /* FIXME i#3544: Put real opcodes here ans make sure ordering matches HW */
+    /* FIXME i#3544: should load/store width be separate opcode or in the instruction
+       flags? */
+    /* FIXME i#3544: Should pseudo instructions be here too? */
```

Comment:
Do these pseudo instructions have a different encoding than others? If not, we probably don't need to add them here separately. We can however add instruction create macros for them.

@derekbruening

Reply:
Okay, I assume the assembler expands the pseudo instructions into the actual machine-recognizable instructions?

If so, then we definitely shouldn't add pseudo opcodes to this list.